lib/common/options_display.c

```diff
@@ -343,7 +346,10 @@ option_list_xml(pcmk__output_t *out, va_list args)
     for (const pcmk__cluster_option_t *option = option_list;
          option->name != NULL; option++) {
 
-        if ((filter == pcmk__opt_context_none) || (filter == option->context)) {
+        if (!pcmk_is_set(option->flags, pcmk__opt_deprecated)
+            && ((filter == pcmk__opt_context_none)
+                || (filter == option->context))) {
```

Comment:
seems to defeat the purpose; easy enough though

Reply:
Basically --all is for humans, XML is for parsers that don't care about information overload. If we had something like --all=true/false it would be fine to default text to false and XML to true (we do something similar with crm_mon).

Reply:
That makes sense and wouldn't be too hard, but probably not worth the complication.

Reply:
I could do `--all=(true|false|default)` at the `crm_attribute` level and require a boolean at the C API level. `crm_attribute` chooses appropriately if `--all=default`.

Not sure if I want to do that or just let XML show everything unconditionally. Lowest priority.